src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -15,103 +15,129 @@ namespace System.Net.Http
     /// </summary>
     internal sealed class DiagnosticsHandler : DelegatingHandler
     {
-        /// <summary>
-        /// DiagnosticHandler constructor
-        /// </summary>
-        /// <param name="innerHandler">Inner handler: Windows or Unix implementation of HttpMessageHandler.
-        /// Note that DiagnosticHandler is the latest in the pipeline </param>
-        public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
-        {
-        }
+        private const string Namespace                      = "System.Net.Http";
+        private const string RequestWriteNameDeprecated     = Namespace + ".Request";
+        private const string ResponseWriteNameDeprecated    = Namespace + ".Response";
+        private const string ExceptionEventName             = Namespace + ".Exception";
+        private const string ActivityName                   = Namespace + ".HttpRequestOut";
+        private const string ActivityStartName              = ActivityName + ".Start";
+        private const string ActivityStopName               = ActivityName + ".Stop";
 
-        internal static bool IsEnabled()
+        private static readonly DiagnosticListener s_diagnosticListener = new("HttpHandlerDiagnosticListener");
+        private static readonly ActivitySource s_activitySource = new(Namespace);
+
+        public static readonly bool IsGloballyEnabled = GetEnableActivityPropagationValue();
```

Comment:
This needs to be static property or it will regress feature switch setting and increases size duo to static ctor dependency

Reply:
Changed to
```c#
public static bool IsGloballyEnabled { get; } = GetEnableActivityPropagationValue();
```

Is that the pattern you had in mind?

Reply:
I think it's better but it will still all the unnecessary dependencies like DiagnosticListener, ActivitySource and others where were not there before. You can check the size regression to see how much it adds

Reply:
Is there a linker switch for `DiagnosticsHandler`?
Or was it capable of removing the code before (it would have had to remove `Activity` and `DiagnosticListener`)?

> You can check the size regression to see how much it adds 

Do you mean just the size of the `System.Net.Http` assembly? Or some other way of looking at trimmed regressions?

Reply:
> Is there a linker switch for DiagnosticsHandler?

Yes, see HttpActivityPropagationSupport at https://github.com/dotnet/runtime/blob/main/docs/workflow/trimming/feature-switches.md

> Do you mean just the size of the System.Net.Http assembly? Or some other way of looking at trimmed regressions?

System.Net.Http will probably show the biggest increase if there is some. You can publish trimmed release build with HttpActivityPropagationSupport property set to false to compare.

Reply:
I updated the `ILLink.Substitutions.xml` to `get_IsGloballyEnabled()`.

Looking at `.\build.cmd -projects .\src\libraries\System.Net.Http\src\System.Net.Http.csproj -os Browser -c Release`
the size of `System.Net.Http.dll` in `browser-wasm\lib\net6.0`, the size dropped from 247.296 to 246.272 bytes after this PR.

I don't know how to test the runtime change with trimming and the feature switch.